contracts/interfaces/IW3CVerifiableCredential.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.16;
+
+import {SmtLib} from "../lib/SmtLib.sol";
+
+/**
+ * @dev IW3CVerifiableCredential
+ */
+interface IW3CVerifiableCredential {
+    struct SubjectField {
+        string key;
+        uint256 value;
+        bytes rawValue;
+    }
+
+    struct Status {
+        string id;
+        string _type;
+        uint64 revocationNonce;
+    }
+
+    struct Schema {
+        string id;
+        string _type;
+    }
+
+    struct State {
+        uint256 rootOfRoots;
+        uint256 claimsTreeRoot;
+        uint256 revocationTreeRoot;
+        uint256 value;
+    }
+
+    struct IssuerData {
+        uint256 id;
+        State state;
+    }
+
+    struct IssuanceProof {
+        string _type;
+        uint256[8] coreClaim;
+        IssuerData issuerData;
+        SmtLib.Proof mtp;
+    }
+
+    struct Credential {
+        uint256 id;
+        string[] context;
+        string[] _type;
+        uint64 expirationDate;
+        uint64 issuanceDate;
+        uint256 issuer;
+        SubjectField[] credentialSubject;
```

Comment:
Curious, I didn't know it's possible that one credential goes to several subjects (array). Can you provide the link to W3C standard? This is just for my understanding.

Reply:
**SubjectFiled** struct describes one field in the **credentialSubject** section. To process credentialSubject from the contact in the user side, we should have: **key**, **value**, **rawValue**. For example, if you have the next credentialSubject:
```json
{
   "balance": 100,
   "name": "BookMarket"
}
```
In the solidity code we will have the next representation:
```
credentualSubject: [
{key: balance, value: 100, rawValue: 0x}
{key: name, value: poseidonHash("BookMarket"), rawValue: 0x143Adsd..
]